domains/transfer/ClaimAutomation/Worker/Automation/ClaimService.cs

```diff
@@ -35,25 +37,40 @@ public async Task Run(CancellationToken stoppingToken)
                 logger.LogInformation("Number of ClaimAutomationArguments for current run: {claimAutomationArguments}", claimAutomationArguments.Count);
                 foreach (var subjectId in claimAutomationArguments.Select(x => x.SubjectId).Distinct())
                 {
-                    var response = await walletClient.GetGranularCertificates(subjectId, stoppingToken);
+                    var hasMoreCertificates = true;
+                    var numberOfFetchedCertificates = 0;
 
-                    if (response == null || !response.Result.Any())
+                    while (hasMoreCertificates)
                     {
-                        logger.LogInformation("No certificates found for {subjectId}", subjectId);
-                        continue;
-                    }
+                        var response = await walletClient.GetGranularCertificates(subjectId, stoppingToken, limit: BatchSize, skip: numberOfFetchedCertificates);
 
-                    logger.LogInformation("Trying to claim {certificates} certificates for {subjectId}", response.Result.Count(), subjectId);
-                    var certificates = response.Result.OrderBy<GranularCertificate, int>(x => shuffle.Next()).ToList();
+                        if (response == null)
+                            throw new ClaimCertificatesException($"Something went wrong when getting certificates from the wallet for {subjectId}. Response is null.");
 
-                    var certificatesGrouped = certificates.GroupBy(x => new { x.GridArea, x.Start, x.End });
+                        if (!response.Result.Any())
+                        {
+                            logger.LogInformation("No certificates found for {subjectId}", subjectId);
+                        }
 
-                    foreach (var cert in certificatesGrouped)
-                    {
-                        var productionCerts = cert.Where(x => x.CertificateType == CertificateType.Production).ToList();
-                        var consumptionCerts = cert.Where(x => x.CertificateType == CertificateType.Consumption).ToList();
-                        logger.LogInformation("Claiming {productionCerts} production certs and {consumptionCerts} consumption certs for {subjectId}", productionCerts.Count, consumptionCerts.Count, subjectId);
-                        await Claim(subjectId, consumptionCerts, productionCerts);
+                        numberOfFetchedCertificates += response.Result.Count();
+                        if (numberOfFetchedCertificates >= response.Metadata.Total)
```

Comment:
Something is fishy here: I expect response.Metadata.Total to decrease with every call because certificates are claimed/moved. But numberOfFetchedCertificates is increased with each loop. When they cross each other, the loop excited, but more certificates might still exist.

The expression for more available certificates, should maybe be something along **if returned page size is less then returned total count**